llvm/lib/Transforms/IPO/ArgumentPromotion.cpp

```diff
@@ -491,10 +491,8 @@ static bool isArgUnmodifiedByAllCalls(Argument *Arg,
                                       FunctionAnalysisManager &FAM) {
   for (User *U : Arg->getParent()->users()) {
 
-    // Bail if we find an unexpected (non CallInst) use of the function.
-    auto *Call = dyn_cast<CallInst>(U);
-    if (!Call)
-      return false;
+    assert(isa<CallBase>(U) && "Expected all users of Function to be CallBase");
+    CallBase *Call = cast<CallBase>(U);
```

Comment:
```suggestion
    auto *Call = cast<CallBase>(U);
```
`cast` performs an implicit `isa` assert.

Reply:
Nice - updated to remove the superfluous assert(), and added some tests for invoke instructions.